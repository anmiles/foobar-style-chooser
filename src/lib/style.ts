import ko from 'knockout';
export { Style };

class Style {
	title: string;
	enabled: (value?: boolean) => boolean;

	constructor(title: string) {
		this.title   = title;
		this.enabled = ko.observable(false);
	}

	toggle() {
		this.enabled(!this.enabled());
	}
}
