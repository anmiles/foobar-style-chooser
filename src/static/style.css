* {
	box-sizing: border-box;
}

html {
	overflow-y: scroll;
}

html, body {
	width: 100%;
	min-height: 100%;
	margin: 0;
	padding: 0;
}

body {
	font-size: 20px;
}

body, input, button {
	font-family: Arial, sans-serif;
}

.main {
    width: 100%;
    min-height: 100%;
    flex-direction: column;
    display: flex;
    gap: 16px;
    padding: 16px;
}

.section {
    flex-direction: row;
    flex-wrap: wrap;
    border: 2px solid currentColor;
    border-radius: 1em;
    position: relative;
    display: grid;
    grid-column-gap: 16px;
    grid-row-gap: 16px;
    padding: 16px;
    grid-template-columns: repeat(12, minmax(0, 1fr));
}

.section {
	color: hsl(20, 50%, 50%);
}

.section.genres {
	color: hsl(120, 50%, 50%);
}

.section.instruments {
	color: hsl(200, 50%, 50%);
}

.section.hidden {
	display: none;
}

.section.output {
	color: hsl(0, 0%, 50%);
}

.section:after {
	content: '';
    display: block;
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    border-radius: inherit;
    background: currentColor;
    filter: opacity(0.3);
}

.button {
    text-transform: capitalize;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    font-size: 1em;
    color: inherit;
    background: rgba(255, 255, 255, 0.25);
    border: 1px solid currentColor;
    border-radius: 0.5em;
    position: relative;
    z-index: 1;
    outline: none;
    padding: 16px;
}

.button:hover {
    background: rgba(255, 255, 255, 0.5);
}

.button.enabled,
.button.enabled:hover {
	background: white;
}

.section.output {
    display: block;
}

.section.output .button {
    background: white;
	width: 100%;
	cursor: default;
}
