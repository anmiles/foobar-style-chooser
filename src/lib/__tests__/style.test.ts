import { Style } from '../style';

const title = 'title';

describe('src/lib/style', () => {
	it('should construct style', () => {
		const style = new Style(title);

		expect(style).toMatchSnapshot();
	});

	describe('toggle', () => {
		it('should invert enabled', () => {
			const style = new Style(title);
			expect(style.enabled()).toEqual(false);

			style.toggle();
			expect(style.enabled()).toEqual(true);

			style.toggle();
			expect(style.enabled()).toEqual(false);
		});
	});
});
