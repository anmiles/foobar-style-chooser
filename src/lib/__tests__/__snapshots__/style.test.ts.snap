// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/lib/style should construct style 1`] = `
Style {
  "enabled": [Function],
  "title": "title",
}
`;
